processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -214,7 +214,26 @@ public File mergeQueryableIndex(
     return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, new BaseProgressIndicator());
   }
 
-  public File mergeQueryableIndex(
+  public File mergeAndCloseMergee(
+      List<QueryableIndex> indexes,
+      boolean rollup,
+      final AggregatorFactory[] metricAggs,
+      File outDir,
+      IndexSpec indexSpec,
+      ProgressIndicator progress
+  ) throws IOException
+  {
+    try {
+      return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, progress);
+    }
+    finally {
```

Comment:
the `true` to Closeables.close should ensure we try to close them all (errors will be skipped).